packages/core/types/@chainlink/index.d.ts

```diff
@@ -1,11 +1,12 @@
 // Declare missing type definitions
 declare module '@chainlink/types' {
-  import type { CacheOptions } from '@chainlink/ea-bootstrap'
+  import type { Cache, RateLimit } from '@chainlink/ea-bootstrap'
   import { cacheWarmer } from "@chainlink/ea-bootstrap"
 
   export interface AdapterContext {
     name?: string
-    cache?: CacheOptions
+    cache?: Cache.CacheOptions
+    rateLimit?: RateLimit.config.Config
```

Comment:
So now each adapter defines it's own `rateLimit` configuration?

Reply:
Not defines it per say, but rather initializes it once on startup and provides it through context rather than re-initializing it on every request.

This was done because of an annoying log message, but is the better way to do it because the config can't change between requests.